### ISSUE TITLE
🔥 Scheduled local notifications with sound in iOS don't display in the foreground

### ISSUE THREAD

**Issue by honeybearhq** [invertase/react-native-firebase#1886]
## Issue

A scheduled local notification won't display in a foregrounded iOS app if a sound is set. 

I schedule a notification as follows:
```
    const notification = new firebase.notifications.Notification()
      .setNotificationId(`Blah`)
      .setBody(`Blah Blah`)
      .setSound('')
    const date = ///
    firebase.notifications().scheduleNotification(notification, {
      fireDate: date.getTime(),
    })
```

Here is my code to display the notification when the app is in the foreground:
```
    firebase
      .notifications()
      .onNotification(notification => {
        firebase.notifications().displayNotification(notification)
      });
```

When the notification is received while the app is in the foreground, I get the following error:
<img width="453" alt="screen shot 2019-02-06 at 10 56 07 am" src="https://user-images.githubusercontent.com/46580806/52366195-fd9e8580-29fd-11e9-8ebb-4b914dd6a523.png">

If i re-set the sound in the code to handle the notification when the app is in the foreground, it works again:

```
    firebase
      .notifications()
      .onNotification(notification => {
        const notificationWithSound = notification.setSound('');
        firebase.notifications().displayNotification(notificationWithSound)
      });
```

---

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
  pod 'Firebase/Core', '~> 5.15.0'
  pod 'Firebase/Auth', '~> 5.15.0'
  pod 'Firebase/Functions', '~> 5.15.0'
  pod 'Firebase/Messaging', '~> 5.15.0'
  pod 'Fabric', '~> 1.7.13'
  pod 'Crashlytics', '~> 3.10.7'
```

#### `AppDelegate.m`:

```objc
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  #ifdef DEBUG
    jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
  #else
    jsCodeLocation = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
  #endif

  [FIRApp configure];
  [RNFirebaseNotifications configure];
  //   ...
  return YES;
}

- (void)application:(UIApplication *)application didReceiveLocalNotification:(UILocalNotification *)notification {
  [[RNFirebaseNotifications instance] didReceiveLocalNotification:notification];
}
```

---

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

---

## Environment

- **Platform that you're experiencing the issue on**:
  - [X] iOS
  - [ ] Android
  - [X] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **If known, the version of the platform are you experiencing the issue on:**
  - `ADD_SOMETHING_HERE` e.g. iOS 10 or Android API 28
- **Operating System:**
  - [X] MacOS, version: `N/A`
  - [ ] Windows, version: `N/A`
  - [ ] Other, please specify: `N/A`
- **Build Tools:**
  - `Xcode 10`
- **`React Native` version:**
  - `0.57.8`
- **`React Native Firebase` library version:**
  - `5.2.1`
- **`Firebase` module(s) you're using that has the issue:**
  - [] **N/A**
  - [X ] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [ ] Cloud **Messaging** (FCM)
  - [X] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] In App Messaging
  - [ ] Indexing
  - [ ] Invites
  - [ ] Instance ID
  - [ ] ML Kit
  - [X] Notifications
  - [ ] Performance Monitoring
  - [ ] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [ ] No
  - [X] Yes, version: `N/A`
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](./../../CONTRIBUTING.md)

**Comment by antaehyeon** [comment]
The same issue. What's the problem?

**Comment by inquedesign** [comment]
I am experiencing this in react-native-firebase 5.3.1 and react-native 0.59.4.  After doing some digging, I found that the notification which is passed into the onNotification() callback has its sound attribute set to null.  I have constructed a local notification and confirmed that it works, but if I give it to the scheduler, it sets the sound property to null before passing it in to the callback.
